src/components/createTestCase/CreateTestCase.tsx

```diff
@@ -573,25 +570,32 @@ const CreateTestCase = () => {
 
   const mapGroups = (
     groupPopulations: GroupPopulation[],
-    results: DetailedPopulationGroupResult
+    populationGroupResults: DetailedPopulationGroupResult[]
   ): DisplayGroupPopulation[] => {
     if (_.isNil(groupPopulations)) {
       return null;
     }
 
-    const gp = groupPopulations.map((groupPop) => ({
-      ...groupPop,
-      populationValues: groupPop?.populationValues?.map((populationValue) => {
-        return {
-          ...populationValue,
-          actual: !!results?.populationResults?.find(
-            (popResult) =>
-              FHIR_POPULATION_CODES[popResult.populationType] ===
-              populationValue.name
-          )?.result,
-        };
-      }),
-    }));
+    const gp = groupPopulations.map((groupPopulation) => {
+      const results = populationGroupResults?.find(
+        (groupResult) => groupResult.groupId === groupPopulation.groupId
+      );
+      return {
+        ...groupPopulation,
+        populationValues: groupPopulation?.populationValues?.map(
+          (populationValue) => {
+            return {
+              ...populationValue,
+              actual: results?.populationResults?.find(
+                (popResult) =>
+                  FHIR_POPULATION_CODES[popResult.populationType] ===
```

Comment:
what will this do for groups with two IPs?

Reply:
I did not think about it. Thanks, Nate for pointing it out.  I just tested. it works fine. that lookup is just to map population names in fhir spec to properties in our models. e.g initial population in spec is `initial-population` but our models have `initialPopulation`. So, even if there are two IPs, they both will have same name and keys.